test/phase/customizability_test.bzl

```diff
@@ -0,0 +1,35 @@
+load(
+    "//scala:providers.bzl",
+    _ScalaRulePhase = "ScalaRulePhase",
+)
+load(
+    "//test/phase:phase_customizability_test.bzl",
+    _phase_customizability_test = "phase_customizability_test",
+)
+load(
+    "//scala:scala.bzl",
+    _make_scala_binary = "make_scala_binary",
+    _make_scala_library = "make_scala_library",
+)
+
+ext_add_phase_customizability_test = {
+    "phase_providers": [
+        "//test/phase:phase_customizability_test",
+    ],
+}
+
+def _add_phase_customizability_test_singleton_implementation(ctx):
+    return [
+        _ScalaRulePhase(
+            phases = [
+                ("-", "coda", "customizability_test", _phase_customizability_test),
+            ],
+        ),
+    ]
+
+add_phase_customizability_test_singleton = rule(
```

Comment:
It'd be nice to move all this into a macro so that consumers don't have to define so many things to customize rules. I think we can do this later.

Reply:
Nvm I don't think you can define rules in a macro :man_facepalming:  
I'll see if there's a different way to make this cleaner from a consumer's standpoint